Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,668 @@
+        navTransform = mathMatrix.matMultiply(a: mathMatrix.matTranslate(v: navLocation), b: locTransform)
+        sendTransform()
+    }
+    
+    func joinPeer(peerId: Int, completionHandler: ((Data, Error?) -> Void)?) {
+        newIceCandidatePeerId = peerId
+        createPeerConnection()
+        inputChannel = self.peerConnection.dataChannel(forLabel: "inputDataChannel", configuration: RTCDataChannelConfiguration())
+        inputChannel.delegate = self
+        createOfferAsync { (offer, error) in
+            self.sendToPeerAsync(peerId: peerId, data: offer, completionHandler: { err in
+                if let completionHandler = completionHandler {
+                    completionHandler(offer, error)
+                }
+            })
+        }
+    }
+    
+    func createPeerConnection() {
+        let defaultPeerConnectionConstraints = RTCMediaConstraints(mandatoryConstraints: nil,
+                                                                   optionalConstraints: ["DtlsSrtpKeyAgreement": "true"])
+        let rtcConfig = RTCConfiguration()
+        let iceServer = RTCIceServer(urlStrings: [Config.turnServer],
+                                     username: Config.username,
+                                     credential: Config.credential,
+                                     tlsCertPolicy: Config.tlsCertPolicy)
+        rtcConfig.iceServers = [iceServer]
+        rtcConfig.iceTransportPolicy = .relay
+        peerConnection = peerConnectionFactory.peerConnection(with: rtcConfig, constraints: defaultPeerConnectionConstraints, delegate: self)
+    }
+    
+    func handlePan(_ recognizer: UIPanGestureRecognizer) {
+        switch recognizer.state {
+        case .began:
+            navOnFingerDown(recognizer: recognizer)
+        case .changed:
+            navOnFingerMove(recognizer: recognizer)
+        case .ended:
+            navOnFingerUp()
+        default:
+            break
+        }
+    }
+    
+    func initGestureRecognizer() {
+        let panGestureRecognizer = UIPanGestureRecognizer(target: self, action: #selector(VideoStreamViewController.handlePan(_:)))
+        panGestureRecognizer.minimumNumberOfTouches = 1
+        panGestureRecognizer.maximumNumberOfTouches = 2
+        self.renderView.addGestureRecognizer(panGestureRecognizer)
+    }
+    
+    func beginTouch(tappedPoint: CGPoint) {
+        fingerDownX = tappedPoint.x
+        fingerDownY = tappedPoint.y
+        downPitch = navPitch
+        downHeading = navHeading
+        downLocation[0] = navLocation[0]
+        downLocation[1] = navLocation[1]
+        downLocation[2] = navLocation[2]
+    }
+    
+    func navOnFingerDown(recognizer: UIPanGestureRecognizer) {
+        let tappedPoint = recognizer.location(in: self.view)
+        beginTouch(tappedPoint: tappedPoint)
+    }
+    
+    func navOnFingerUp() {
+        twoFingersWereUsed = false
+    }
+    
+    func navOnFingerMove(recognizer: UIPanGestureRecognizer) {
```

Comment:
does this being present still mean we support both the device motion and finger movement panning at the same time?

Reply:
Yes, I made sure that rotation and translation worked with the finger. Michelle made sure rotation worked with the accelerometer. The features work together, and we thought for accessibility reasons it makes sense to have both.